# Тема 7. Полезные инструменты. Обработка исключений.
# Решить с помощью генераторов списка.
#
# 4. Написать функцию которая принимает на вход число от 1 до 100. Если число равно 13, функция поднимает
# исключительную ситуации ValueError иначе возвращает введенное число, возведенное в квадрат.
# Далее написать основной код программы. Пользователь вводит число. Введенное число передаем параметром в написанную
# функцию и печатаем результат, который вернула функция. Обработать возможность возникновения исключительной ситуации,
# которая поднимается внутри функции.


while True:
    try:
        number: int = int(input('Введите целое число от 0 до 100: '))
        break
    except ValueError:
        print('Введите число!')
        continue


def number_random(i):
    try:
        if i == 13:
            raise ValueError
        if i < 0 or i > 100:
            raise Exception
    except ValueError as e:
        return 'Исключительная ситуация - число равно 13', e
    except Exception as e:
        return 'Неизвестная исключительная ситуация', e
    else:
        return i ** 2


result = number_random(number)

print(f'Число, возведенное в квадрат - {result}')
